src/commands/synthetics/deploy-tests-lib.ts

```diff
@@ -0,0 +1,93 @@
+import {getCommonAppBaseURL} from '../../helpers/app'
+
+import {APIHelper, EndpointError, formatBackendErrors, getApiHelper} from './api'
+import {DeployTestsCommandConfig, LocalTriggerConfig, MainReporter, ServerTest, TestConfig} from './interfaces'
+import {findUniqueLocalTestDefinition} from './local-test-definition'
+import {getTestConfigs} from './test'
+import {isLocalTriggerConfig} from './utils/internal'
+
+export const deployTests = async (reporter: MainReporter, config: DeployTestsCommandConfig): Promise<void> => {
+  const api = getApiHelper(config)
+
+  reporter.log('Deploying tests...\n\n')
+
+  const testConfigFromFile: TestConfig = {
+    tests: await getTestConfigs(config, reporter),
+  }
+
+  if (config.publicIds.length > 0) {
+    testConfigFromFile.tests = testConfigFromFile.tests.filter(
+      (test) =>
+        isLocalTriggerConfig(test) &&
+        test.local_test_definition.public_id &&
+        config.publicIds.includes(test.local_test_definition.public_id)
+    )
+  }
+
+  const publicIds = new Set(
+    testConfigFromFile.tests.flatMap((test) =>
+      isLocalTriggerConfig(test) && test.local_test_definition.public_id ? [test.local_test_definition.public_id] : []
+    )
+  )
```

Comment:
Why do you need a `flatMap` and a `Set` here? What different between `publicIds` and `testConfigFromFile.tests`?

Reply:
I wanted to follow the exact same logic [as here](https://github.com/DataDog/datadog-ci/blob/82bbf9832259a05efdc9b711bf4496bfb02d36a1/src/commands/synthetics/multilocator.ts#L91) where we iterate over unique `publicIds`, then use `findUniqueLocalTestDefinition()` to enforce the public ID to be unique in the file.

But it's true that we actually don't need it, since `findUniqueLocalTestDefinition()` will enforce the uniqueness already.